server/src/main/java/org/elasticsearch/http/AbstractHttpServerTransport.java

```diff
@@ -387,9 +401,15 @@ protected void serverAcceptedChannel(HttpChannel httpChannel) {
      */
     public void incomingRequest(final HttpRequest httpRequest, final HttpChannel httpChannel) {
         httpClientStatsTracker.updateClientStats(httpRequest, httpChannel);
+        final RequestTrackingHttpChannel trackingChannel = httpChannels.get(httpChannel);
         final long startTime = threadPool.rawRelativeTimeInMillis();
         try {
-            handleIncomingRequest(httpRequest, httpChannel, httpRequest.getInboundException());
+            if (trackingChannel != null) {
+                // The channel may not be present if the close listener (set in serverAcceptedChannel) runs before this method because the
+                // connection closed early
+                trackingChannel.incomingRequest();
+            }
+            handleIncomingRequest(httpRequest, trackingChannel != null ? trackingChannel : httpChannel, httpRequest.getInboundException());
```

Comment:
Why are we processing the request if we are in the edge case that the tracking channel was closed and removed?

Reply:
Good point, returning in that case.

Reply:
Unfortunately, there is a unit test,[testMalformedRequestDispatchedNoAuthn](https://github.com/elastic/elasticsearch/blob/0e95bb4ef7ffd39aa7f88526df0f7c05ce2138c1/x-pack/plugin/security/src/test/java/org/elasticsearch/xpack/security/transport/netty4/SecurityNetty4HttpServerTransportTests.java#L676), that expects the request to still be dispatched, in that case.